kucher/view/main_window/register_view_widget/dialog_export.py

```diff
@@ -0,0 +1,197 @@
+
+        asyncio.get_event_loop().create_task(executor())
+
+
+class ImportDialogWindow(QWidget):
+
+    def __init__(self, parent, registers):
+        self.parent = parent
+        self.registers = registers
+        super().__init__()
+        self.setGeometry(10, 10, 1000, 700)
+        self._saveFileDialog()
+
+    def _saveFileDialog(self):
+        self.fileName, _ = QFileDialog.getOpenFileName(self, 'Import configuration file', '',
+                                                       'YAML Files (*.yml *.yaml);;All Files (*)')
+        if not self.fileName:
+            return
+        try:
+            self._file = open(self.fileName, 'r')
+        except IOError:
+            _logger.exception(f'File {self.fileName} could not be open')
+            return
+
+        # Check before writing any register if they are available on the device and within the permitted range
+        self.imported_registers = yaml.safe_load(self._file)
+        try:
+            for reg_check in self.registers:
+                if reg_check.name in self.imported_registers.keys():
+                    if not reg_check.mutable:
+                        show_error('Import failed',
+                                   f'Cannot import {self.fileName}',
+                                   f'This parameter cannot be modified on this device:\n{reg_check.name}',
+                                   self.parent)
+                        self._file.close()
+                        _logger.exception(f'Import failed : This parameter cannot be modified on this '
+                                          f'device: {reg_check.name}')
+                        return
+
+                    elif reg_check.has_min_and_max_values and reg_check.type_id != Register.ValueType.BOOLEAN:
+                        if not (reg_check.min_value <= self.imported_registers[reg_check.name] <= reg_check.max_value):
+                            show_error('Import failed',
+                                       f'Cannot import {self.fileName}',
+                                       f'This parameter value is outside permitted range:\n{reg_check.name}',
+                                       self.parent)
+                            self._file.close()
+                            _logger.exception('Import failed :This parameter value is outside '
+                                              f'permitted range: {reg_check.name}')
+                            return
+
+        except Exception as ex:
+            show_error('Import failed',
+                       f'Cannot import {self.fileName}',
+                       f'Reason: {str(ex)}',
+                       self.parent)
+            _logger.exception(f'Could not write registers: {ex!r}')
+            self._file.close()
+            return
+
+        self._file.close()
+
+        async def executor():
+            for _attempt in range(3):
+                try:
+                    await self._writeRegisters()
```

Comment:
The retry logic should be implemented on a per-register basis, not for the whole set.